### PR TITLE
build: update typescript version

### PR DESCRIPTION
**Pull request by Feiyang1**
Part of https://github.com/firebase/firebase-js-sdk/pull/1549
Update dependencies in preparation to enabling auto-updates.

### CODE REVIEW

File: packages/firestore/src/util/misc.ts [hunk 1]
```diff
@@ -24,14 +24,14 @@ export type EventHandler<E> = (value: E) => void;
...
-export type AnyJs = null | undefined | boolean | number | string | object;
+export type AnyJs = unknown;
 
 /**
  * `Unknown` is a stand-in for Typescript 3's `unknown` type. It is similar to
  * `any` but forces code to check types before performing operations on a value
  * of type `Unknown`. See: https://blogs.msdn.microsoft.com/typescript/2018/07/30/announcing-typescript-3-0/#the-unknown-type
  */
-export type Unknown = null | undefined | {} | void;
+export type Unknown = unknown;
```
Comment (Feiyang1) [inline packages/firestore/src/util/misc.ts]: @mikelehen Can Unknown and AnyJs be replaced by 'unknown'? If it sounds right to you, I will make a another PR to remove AnyJs and Unknown and replace all occurrences with `unknown`

File: packages/database/src/core/util/Tree.ts [hunk 2]
```diff
@@ -155,7 +155,7 @@ export class Tree<T> {
    * @return {boolean} true if the action callback returned true.
    */
   forEachAncestor(
-    action: (tree: Tree<T>) => void,
+    action: (tree: Tree<T>) => any,
```
Comment (mikelehen) [inline packages/database/src/core/util/Tree.ts]: Can this be `unknown`?

File: packages/firestore/src/util/misc.ts (hunk 1 above)
Comment (mikelehen) [reply packages/firestore/src/util/misc.ts]: Yes, that sounds right to me.  But let's do the massive search/replace in a separate PR (I can take care of it if you like).  For now, what you've done here looks good.  You could add a TODO to delete AnyJs and Unknown if you like.

File: packages/firestore/test/util/test_platform.ts [hunk 3]
```diff
@@ -87,7 +87,11 @@ export class FakeWindow {
  * `Document` fake that implements the `visibilitychange` API used by Firestore.
  */
 export class FakeDocument {
-  private _visibilityState: VisibilityState = 'unloaded';
+  /**
+   * unload is no more a valid member of VisibilityState, consider changing it to a valid value.
+   * https://developer.mozilla.org/en-US/docs/Web/API/Document/visibilityState
+   */
+  private _visibilityState: VisibilityState = 'unloaded' as VisibilityState;
```
Comment (mikelehen) [inline packages/firestore/test/util/test_platform.ts]: I think this was just a plain mistake.  Can you remove the comment you added and just change 'unloaded' to 'hidden'?  Thanks!

File: packages/messaging/src/helpers/array-buffer-to-base64.ts [hunk 4]
```diff
@@ -17,7 +17,7 @@
 
 function toBase64(arrayBuffer: ArrayBuffer | Uint8Array): string {
   const uint8Version = new Uint8Array(arrayBuffer);
-  return btoa(String.fromCharCode.apply(null, uint8Version));
+  return btoa(String.fromCharCode.apply(null, Array.from(uint8Version)));
```
Comment (mmermerkaya) [inline packages/messaging/src/helpers/array-buffer-to-base64.ts]:
```suggestion
  return btoa(String.fromCharCode(...uint8Version));
```

Comment (Feiyang1) [reply packages/messaging/src/helpers/array-buffer-to-base64.ts]: Nice.

File: packages/firestore/src/util/misc.ts (hunk 1 above)
Comment (Feiyang1) [reply packages/firestore/src/util/misc.ts]: Sure. I will create a separate PR for this.

File: packages/database/src/core/util/Tree.ts (hunk 2 above)
Comment (Feiyang1) [reply packages/database/src/core/util/Tree.ts]: Done.

File: packages/firestore/test/util/test_platform.ts (hunk 3 above)
Comment (Feiyang1) [reply packages/firestore/test/util/test_platform.ts]: Done.

**Pull request merged**